### PR TITLE
(FACT-2815) Added timing for cached facts

### PR DESCRIPTION
**Pull request by sebastian-miclea**
Updated --timing option to also show time needed to read from cached. Before, cached facts were skipped.

### CODE REVIEW

File: spec/facter/cache_manager_spec.rb [hunk 1]
```diff
@@ -135,6 +136,32 @@
...
+
+      it 'returns cached external facts' do
+        allow(fact_groups).to receive(:get_fact).with('os').and_return(nil)
+        allow(fact_groups).to receive(:get_fact).with('my_custom_fact').and_return(nil)
+        allow(fact_groups).to receive(:get_fact).with('ext_file.txt').and_return(external_fact)
+        allow(Facter::Util::FileHelper).to receive(:safe_read).with(File.join(cache_dir, 'ext_file.txt'))
+                                                              .and_return(cached_external_fact)
+        allow(File).to receive(:mtime).with(File.join(cache_dir, 'ext_file.txt')).and_return(Time.now)
+
+        cache_manager.resolve_facts(searched_facts)
```
Comment (Filipovici-Andrei) [inline spec/facter/cache_manager_spec.rb:161]: You could add an empty line to make the separation between the Arrange, Act and Assertion parts of the unit test.

**Comment by Filipovici-Andrei** [comment]
Does this PR resolve the inconsistent timing issue of cached facts

**Review by IrimieBogdan — APPROVED** [review]

File: spec/facter/cache_manager_spec.rb (hunk 1 above)
Comment (sebastian-miclea) [reply spec/facter/cache_manager_spec.rb:161]: Updated.

File: lib/facter/framework/benchmarking/timer.rb [hunk 2]
```diff
@@ -7,11 +7,12 @@ module Framework
...
       class Timer
         class << self
-          def measure(fact_name)
+          def measure(fact_name, prefix_message = '')
             if Options[:timing]
               time = Benchmark.measure { yield }
 
-              puts "fact `#{fact_name}`, took: #{time.format('%r')} seconds"
+              log = "fact '#{fact_name}', took: #{time.format('%r')} seconds"
+              puts "#{prefix_message} #{log}"
```
[the remark this answers is not in the document]
Comment (sebastian-miclea) [reply lib/facter/framework/benchmarking/timer.rb]: Updated.

**Comment by IrimieBogdan** [comment]
> Does this PR resolve the inconsistent timing issue of cached facts?

It should https://tickets.puppetlabs.com/browse/FACT-2815?focusedCommentId=764361&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-764361

**Review by Filipovici-Andrei — APPROVED** [review]

**Pull request merged**